src/pyenphase/firmware.py

```diff
@@ -16,37 +27,48 @@ class EnvoyFirmware:
         "_firmware_version",
         "_serial_number",
         "_part_number",
+        "_timeout",
     )
 
-    def __init__(self, _client: httpx.AsyncClient, host: str) -> None:
+    def __init__(
+        self,
+        _client: httpx.AsyncClient,
+        host: str,
+        timeout: float | httpx.Timeout | None = None,
+    ) -> None:
         """Initialize the Envoy firmware version."""
         self._client = _client
         self._host = host
         self._firmware_version: str | None = None
         self._serial_number: str | None = None
         self._part_number: str | None = None
+        self._timeout = timeout or LOCAL_TIMEOUT
```

Comment:
Thinking about this some more, I think we should not allow a timeout to be passed here it adds more complexity and the potential for more to go wrong

Reply:
Would think it makes it more consistent. If, and only if, caller specifies a timeout for Envoy setup, it is passed on from Envoy to here and both Envoy and Firmware use the same, otherwise the Envoy default is passed on.

Defaults remain the same as now, but if users decides to use something else it is used consistent throughout. 

From Envoy __init__
```py
        self._timeout = timeout or LOCAL_TIMEOUT
        self._client = client or httpx.AsyncClient(
            verify=NO_VERIFY_SSL_CONTEXT
        )  # nosec
        self.auth: EnvoyAuth | None = None
        self._host = host
        self._firmware = EnvoyFirmware(self._client, self._host, self._timeout)
        self._supported_features: SupportedFeatures | None = None
```